back/src/Application/Buying/Handler/CreateBuyingCommandHandler.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Application\Buying\Handler;
+
+use App\Application\Buying\Command\CreateBuyingCommand;
+use App\Domain\Apiary\Repository\ApiaryRepositoryInterface;
+use App\Domain\Buying\Buying;
+use App\Domain\Buying\Repository\BuyingRepositoryInterface;
+
+class CreateBuyingCommandHandler
+{
+    public function __construct(
+        private readonly BuyingRepositoryInterface $buyingRepository,
+        private readonly ApiaryRepositoryInterface $apiaryRepository,
+    ) {
+    }
+
+    public function __invoke(CreateBuyingCommand $command): Buying
+    {
+        $payload = $command->payload;
+        $user = $command->user;
+        // $apiary = $command->apiary;
+
+        $apiary = $this->apiaryRepository->getOneByUid($payload->apiary);
```

Comment:
```suggestion
        /** @var Apiary|null $apiary */
        $apiary = null;
```
ça claque une erreur ici si tu cherche une apiary avec un $payload->apiary null. On initie la variable ici à la place, et on teste ensuite si le payload carry un id d'apiary, puis on getOneByUid comme tu l'as fais ci-dessous 👍 👌